include/z64collision_check.h

```diff
@@ -76,23 +76,23 @@ typedef struct {
 } ColliderInitToActor; // size = 0x08
 
 typedef struct {
-    /* 0x00 */ u32 dmgFlags; // Toucher damage type flags.
+    /* 0x00 */ u32 dmgFlags; // Damage types dealt by this collider element as AT.
     /* 0x04 */ u8 effect; // Damage Effect (Knockback, Fire, etc.)
-    /* 0x05 */ u8 damage; // Damage or Stun Timer
-} ColliderElementTouch; // size = 0x08
+    /* 0x05 */ u8 damage; // Damage
+} ColliderElementDamageInfoAT; // size = 0x08
```

Comment:
Seems like a potentially good idea to remove "damage" from these field names since that information is now contained within the wrapping struct

`dmgFlags` -> `typeFlags` possibly-- both the old and new comments refer to it as a "damage type". 
(and less importantly, thats also what it was originally called: `at_type`, see en_arrow)

`damage` -> maybe `amount`-- This sort of helps in the cases where this value isnt even a damage value. But it will still be implied by its wrapper struct having "damage" in it. But either way I think it removes some redundancy.

with these changes you would have:
`this->collider.elem.atDmgInfo.typeFlags`
`this->collider.elem.atDmgInfo.amount`

Im not 100% sold on this yet, just an idea. 
I have a bone to pick with "effect" as well, but will leave that for another time.

Reply:
also this change doesn't necessarily need to happen right now, I think I agree with the overall change that this PR introduces. just some more changes that can be made on top of it if we like it

Reply:
I like your suggestion better than what it is currently
I'll try to think of alternatives

We could also name `atDmgInfo.dmgFlags` differently from `acDmgInfo.dmgFlags` to help readability (those two only differ by one letter in ac/at)
`atDmgInfo.dmgFlags` -> `atDmgInfo.dealtTypes`
`acDmgInfo.dmgFlags` -> `acDmgInfo.sensibleTypes`
(bad names but to give an idea)

Reply:
I like your suggestion better than what it is currently
I'll try to think of alternatives

We could also name `atDmgInfo.dmgFlags` differently from `acDmgInfo.dmgFlags` to help readability (those two only differ by one letter in ac/at)
`atDmgInfo.dmgFlags` -> `atDmgInfo.dealtTypes`
`acDmgInfo.dmgFlags` -> `acDmgInfo.sensibleTypes`
(bad names but to give an idea)

Reply:
`atDmgInfo.attackTypes` `acDmgInfo.weakTypes` maybe? can't really think of anything super great.

Other ideas: 

* `attackFlags` and `immuneFlags` and just explain that 0s are immune, 1s are vulnerable.
* `attackFlags` and `vulnerabilityFlags`

Reply:
(brainstorming)

I like "vulnerabilities", could be shortened to `vuln`

I think Types is more informative than Flags but either may work, unsure

`atDmgInfo.dealTypes`
`acDmgInfo.vulnTypes`

`attack Types/Flags` would be expanding our assumed meaning for AT to attack, then AC could expand to check
`atDmgInfo.attackTypes`
`acDmgInfo.checkTypes`